### ISSUE TITLE
[SDK-CORE] SDK-Core V1

### ISSUE THREAD

**Issue by 0xdavinchee** [superfluid-finance/protocol-monorepo#1326]
tl;dr
- redefining the purpose of the SDK-Core in the future from first principles
  - imo, we are eliminating friction for devs
- seek leverage, do not introduce new concepts in an SDK, make the interface familiar to what people are used to

https://viem.sh/docs/benchmarks.html
Benchmarks aside, I think a topic of discussion is the future of the SDK-core and taking a holistic approach in assessing the core (hehe) purpose of the SDK-core.

In my opinion, this may **NOT** be worth it for the current SDK-core (unless there is a really strong reason to given resource constraints), but is something to consider in future versions or "the final version" for the V1 Protocol.

We may want to reevaluate the purpose of the SDK-core given the new tools that are being released.

When designing the next version of the SDKs or anything for that matter, we should consider:
1. what is the purpose of this package? is it for hackathon devs? wave pool devs? devs seeking funding at some point?
2. who is it for? who is the end consumer?
3. what are the underlying frameworks used to build this? who is building it? do they have incentive to carry this forward?

In my opinion, what we are essentially doing with the SDK is eliminating as much friction as possible, that is:
- [ ] packaging the extra step of finding our contract addresses (done w/ metadata)
- [ ] gathering the ABIs and
- [ ] providing a plug-n-play interface for developers

Ideally, they should be able to read a single page of docs and create a stream (which I think is the case right now) and no more than a Saturday to understand how to use the 20% of the SDK to accomplish 80% of what they need.

We should reduce friction by leveraging prior knowledge on how to use specific frameworks rather than introduce new ways (requires learning) of interacting with smart contracts, this enables us to leverage a % of developers who have  (web3js, ethers, viem) to quickly iterate on ideas.

**Comment by hellwolf** [comment]
Latest idea related to this issue: https://github.com/superfluid-finance/protocol-monorepo/issues/1575
